hubspot3/base.py

```diff
@@ -95,7 +111,31 @@ def _get_path(self, subpath):
 
     def _prepare_request_auth(self, subpath, params, data, opts):
         if self.api_key:
-            params["hapikey"] = params.get("hapikey") or self.api_key
```

Comment:
@Jenselme 

I would be happy to have your opinion on this part of the code. I am not sure why we made this change and I don't know yet if It should be removed.

Reply:
It feels weird and I'm not sure I understand what it does (mostly why we need to catch `AttributeError`). Did you try to remove these changes and see if it still works with our use cases?

Reply:
After few tests, it seems to work nice without those changes. We are always using the `api_key` so it seems ok.

The only thing that scare me is this comment regarding `urlencode` but I cannot remember now a reason to this comment.

I think that it is preferable to remove it now and see later if we really need to update this method. Right now I am unable to justify this change to the owner of the library.